### DISCUSSION TITLE
Should the new examples have request level logging on by default?

### DISCUSSION THREAD

**Discussion by ChaseLewis** [tokio-rs/axum#1618] [General]
I noticed in the new logging examples that were updated a few days  ago the tracing subscriber didn't work with request level logging. Just logging in the application.

```
    tracing_subscriber::registry()
        .with(
            tracing_subscriber::EnvFilter::try_from_default_env()
                .unwrap_or_else(|_| "{package_name}=debug".into()),
        )
        .with(tracing_subscriber::fmt::layer())
        .init();
```

where package_name is the name field under the [package] tag in the Cargo.toml. For request logging you need.

```
    tracing_subscriber::registry()
        .with(
            tracing_subscriber::EnvFilter::try_from_default_env()
                .unwrap_or_else(|_| "package_name=debug,tower=debug".into()),
        )
        .with(tracing_subscriber::fmt::layer())
        .init();
```

You have to include tower cargo name or no logging.

**Comment by davidpdrsn** [comment]
It should have `tower_http=debug` as well. If some examples don't that's a mistake. Thanks for noticing! Do you wanna submit a PR perhaps?

**Reply by ChaseLewis** [reply]
Yeah i'll update them
